### ISSUE TITLE
Temperature is configured with hot-end & bed but set independently.

### ISSUE THREAD

**Issue by JoshMcCullough** [OctoPrint/OctoPrint#216]
In the settings dialog, we create temperature "profiles" which specify a name, hot-end temperature and bed temperature. But there is no place that allows us to choose that profile to set both the hot-end and bed temperature to the "profile" at the same time.

**Comment by tekkydave** [comment]
I realise this is an old issue and only cosmetic but could it be looked at. It's pointless having two temperatures in the settings when you still have to select them separately on the temperatures tab. I'm surprised no-one else has brought it up since 2013.

**Comment by nophead** [comment]
But I often heat the nozzle without the bed to change filament.

**Comment by tekkydave** [comment]
That's a good point. Maybe the settings need to be a single temperature rather than a pair.
Alternatively I would like to select a profile for the filament I'm using so it sets both temperatures. I may just implement custom controls to achieve the same result.

**Comment by nophead** [comment]
When I am printing my gcode sets the temperatures, so I only use manual control for changing filament, or using the bed heater to cure glue / dry things out, etc. I never manually set up for printing, so have never needed to set them as a pair.

**Comment by tekkydave** [comment]
We will all have different workflows. I usually get the bed and hotend to working temp manually before starting a print. When you pick a hotend temperature there are 2 temps listed in the dropdown when you only need one.

**Comment by light24bulbs** [comment]
Right now it takes four clicks to set up temperatures for a material.  I have to use the drop down on the bed and the extruder to set the termpatures.  In the empty space under this dialog, I would like buttons that select the material and set that as the temperature.  It would be cool if they had a check mark or turned into a btn-success(green) when they were selected.  

Here is my terrible artists impression.  

![octoprintedited](https://cloud.githubusercontent.com/assets/581906/25032045/e47def60-2085-11e7-9e3f-0b1ceb94d97a.png)

**Comment by FedericoNembrini** [comment]
Old request, but I think I can contribute.

![OctoPrint - #216 - v0 1](https://user-images.githubusercontent.com/25205086/83976764-5769d500-a8fc-11ea-8f6e-83fcc3d76abd.gif)

Could something like this be fine? (Haven't added styles yet)

**Comment by light24bulbs** [comment]
@FedericoNembrini I think that's exactly how we are imagining it.  In the interest of keeping the UI uncluttered, I think just buttons with the profile names on them would also work.  Temperature could be in a tooltip.  But what you built also looks great.  

I think a good name for the box that contains the profile buttons could be "preheat".  That way it uses the same terminology and worjs exactly the same as many printer controllers, like Prusas.

**Comment by FedericoNembrini** [comment]
@light24bulbs I'm working on it. I will post updates as soon as I have fixed the last things.

For the layout with only the buttons and profile names: this could create inconsistency in the UI because the buttons could all be of different size, unless we decide to set them with a fixed width, risking having to truncate the profile name.
Example:
![image](https://user-images.githubusercontent.com/25205086/84076473-15ad5d00-a9d6-11ea-9d07-2e8bb84c3736.png)

Once the first layout is finished I will try the one you suggested. In case you have other considerations don't hesitate to let me know.

**Comment by FedericoNembrini** [comment]
Final Version:
![OctoPrint - #216 - v1 0](https://user-images.githubusercontent.com/25205086/84389354-35bb6700-abf6-11ea-8633-b05a092d18f3.gif)

The layout with buttons only is a bit more complex because it requires to add a virtual grouping. (To display row of fixed numbers of elements). IMHO the "table" version is the easiest solution and require very few changes.

As you can see from the GIF, **the profile is applied to all the extruders present**. It is applied to the chamber (from what I know at the moment, in the profiles it is not possible to specify it so it will always be set to off) and for the heated bed.

The only doubt I have left is: if the printer has multiple extruders, will the power supply manage to deliver enough power to bring the temperature for all of them at the same time? We need to think about this?

I look forward to feedback. Thanks.

**Comment by JoshMcCullough** [comment]
Regarding the "table" vs. buttons options, I think it can be done with a drop-down button below the existing Target inputs/buttons, crude mockup of this button:

```
[ Use Preset    ▼ ]
| PLA 1           |
| PLA 2           |
| ...             |
| ABS             |
-------------------
```

The temps could be displayed in a tooltip, on hover of each option.

And the presets should really allow for setting all things, sound like it does not currently?

**Comment by jneilliii** [comment]
What about something like [Terminal Commands Extended](https://plugins.octoprint.org/plugins/terminalcommandsextended/) for the temperature tab?

**Comment by FedericoNembrini** [comment]
@JoshMcCullough you mean under the table?
For what I have seen the profiles have a property called "chamber" but it is always set to zero, with no possibility of modification. I might be wrong.

@jneilliii if you are referring to the layout, it can be done, personally I prefer the version with the table as it requires few changes and the risk of error is really low. If, on the other hand, you are referring to the idea of plugins itself, I would say that this should be a basic functionality since it is the same OctoPrint that allows you to specify presets.

**Comment by jneilliii** [comment]
My point was definitely relative to this being a plugin rather than built into core. Having as a plugin allows for the user to choose rather they want it or not. Referencing settings in core octoprint is a non issue.

**Comment by JoshMcCullough** [comment]
@FedericoNembrini I am revising my statement a bit. IMO the table in your screenshots, while useful, is a bit too much on the UI. I'm suggesting a very simple button. Hacked this in...

![image](https://user-images.githubusercontent.com/4859660/84534375-c8700a80-acb7-11ea-914c-f1fc17594e8a.png)

**Comment by FedericoNembrini** [comment]
@jneilliii I understand your point of view, but IMHO why have the ability to create profiles if I can't apply them (i.e. without going to select the temperatures individually?). I think it's something that should be included in the core.

@JoshMcCullough understood.

**Comment by jneilliii** [comment]
I guess it depends on implementation. As mentioned before several people have different workflows, so using the drop-down to set temps in existing UI makes sense and doesn't clutter the interface, but also introduces other questions. Do the commands spawn off simultaneously or do you get to choose options like wait for bed temperature to be reached prior to heating up nozzle to prevent oozing related issues, etc? I think that was the reasoning behind the current implementation best I can tell. 

Would it make more sense to leave the UI as is, and just change the behavior? For example click either of the presets in any of the temp drop downs and it automatically changes the others to match the same profile and has some hierarchy setting to know which temp to wait for, if at all, before heating up the other. I could see something like Chamber and Bed temps wanting to be reached prior to the nozzle even starting to heat up.

**Comment by FedericoNembrini** [comment]
I agree on what you say. From my point of view I was aiming for something relatively simple to get a profile (set of temperatures) applied quickly without too many controls or ordering settings or anything else.

With the implementation of a plugin there would be more customization on the settings, consequently more workflows could be considered.

Idk for change in behaviour, it could be misunderstood.

**Comment by JoshMcCullough** [comment]
@jneilliii I agree with the setting of other temps within the same profile when you make the initial temp selection from _any_ temp dropdown. But it's only useful, I think, if the temps were all previously set to _off_. I hope that makes sense  -- still need to support tweaking temps even after you select the profile to use for a given print-run.

**Comment by light24bulbs** [comment]
@JoshMcCullough @jneilliii  I feel that could be a bit obtuse, as the user would not expect a dropdown for one temperature to set the temperature for the other field.  Also, you're changing the behavior of an existing control for real world robots in which heating is the most critical and dangerous function.  

 I feel a seperate button under the field "Preheat" is much more clear.  Again, that is exactly how most built-in 3d printer interfaces work.

**Comment by JoshMcCullough** [comment]
@light24bulbs Sure, I think this is probably the better option: https://github.com/OctoPrint/OctoPrint/issues/216#issuecomment-643419856

**Comment by FedericoNembrini** [comment]
Update

![localhost_5000_](https://user-images.githubusercontent.com/25205086/85621820-2c0a1880-b666-11ea-9263-d582e769e2b7.png)
![localhost_5000_ (1)](https://user-images.githubusercontent.com/25205086/85621808-24e30a80-b666-11ea-8c58-a911f6825f9e.png)

**Comment by foosel** [comment]
Now make it non primary and a right floated icon button like the file list settings and similar and we have a deal 👍

**Comment by FedericoNembrini** [comment]
I don't know if I understand correctly, but do you mean this?

![localhost_5000_](https://user-images.githubusercontent.com/25205086/85860888-c03bc300-b7bf-11ea-97de-5844235f46c5.png)

**Comment by foosel** [comment]
Almost:

![image](https://user-images.githubusercontent.com/83657/85863950-2b879400-b7c4-11ea-8ecb-2082a7f23f20.png)

**Comment by FedericoNembrini** [comment]
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/25205086/85923002-b087b180-b887-11ea-8d56-6a135c69c443.gif)

If this is correct, I can proceed with the pull request.

**Comment by jneilliii** [comment]
That looks right to me, based on @foosel's screenshot.

**Comment by foosel** [comment]
Yep 👍 pull request against the `maintenance` branch please, then I can still get it into the 1.4.1 release, first RC should land early next week.

**Comment by foosel** [comment]
Implemented by #3617, to be released as part of 1.4.1

**Comment by foosel** [comment]
1.4.1 has just been released
